crates/storage/db/src/snapshot/mask.rs

```diff
@@ -0,0 +1,70 @@
+use crate::table::Decompress;
+
+/// Helper struct to select column values from a snapshot row
+#[derive(Debug)]
+pub struct Mask<T, J = (), K = ()>(std::marker::PhantomData<(T, J, K)>);
+
+/// Mask for Header snapshot segment.
+pub type HeaderMask<T, J = (), K = ()> = Mask<T, J, K>;
+
+/// Mask for Receipt snapshot segment.
+pub type ReceiptMask<T, J = (), K = ()> = Mask<T, J, K>;
+
+/// Mask for Transaction snapshot segment.
+pub type TransactionMask<T, J = (), K = ()> = Mask<T, J, K>;
+
+/// Column mask for one value.
+pub trait ColumnMaskOne {
+    /// First column value
+    type T: Decompress;
+    /// Mask to obtain desired values. Should be in the same order as snapshot creation.
+    const MASK: usize;
+}
+
+/// Column mask for two values.
+pub trait ColumnMaskTwo {
```

Comment:
can we get a few more docs about what this actually does, and what all of this represents.

having a hard time deciphering what a mask, column, row is in this context.